### PR TITLE
fix: various forecasting fixes

### PR DESCRIPTION
**Pull request by egordm**
This pull request introduces several improvements and fixes across the forecasting pipeline, focusing on enhanced data handling, better model configuration, and improved feature engineering. The most significant changes include updates to the example pipeline to use a more sophisticated model, improvements to the scaler and lag transforms, enhancements to model serialization, and bug fixes in loss functions.

**Example Pipeline and Model Usage:**

* The example script (`configuring_model_pipeline_example.py`) now demonstrates a more realistic forecasting workflow using synthetic time series data, the `GBLinearForecaster` (instead of the constant median forecaster), and includes plotting of predictions. It also shows how to set hyperparameters and use the improved scaler and lag transforms. [[1]](diffhunk://#diff-a86af6102d8e37cec4e22e70ecd7cad2814da09ec566bed79b1e24c7ce67f25dL39-R47) [[2]](diffhunk://#diff-a86af6102d8e37cec4e22e70ecd7cad2814da09ec566bed79b1e24c7ce67f25dL54-R104) [[3]](diffhunk://#diff-a86af6102d8e37cec4e22e70ecd7cad2814da09ec566bed79b1e24c7ce67f25dL84-R143)

**Feature Engineering and Data Transformation:**

* `ScalerTransform` now supports specifying a subset of columns to scale, and its implementation has been updated to only apply scaling to those columns. [[1]](diffhunk://#diff-bcef4bff5384f5e656a93c8315e1b5fe1fc7a2c905fb99162f01c9a6fc7b4de9R69-R72) [[2]](diffhunk://#diff-bcef4bff5384f5e656a93c8315e1b5fe1fc7a2c905fb99162f01c9a6fc7b4de9L98-R113)
* `VersionedLagTransform` has been fixed to ensure lagged features do not extend the dataset's time range, preventing the creation of timepoints outside the original data. [[1]](diffhunk://#diff-a3752ed056a5a899b55d9a36260ce141009201b17c8af4cdd82c4a3774b4120cL83-R91) [[2]](diffhunk://#diff-a3752ed056a5a899b55d9a36260ce141009201b17c8af4cdd82c4a3774b4120cR140-R150)

**Forecasting Model Improvements:**

* The `ForecastingModel` now ensures the target column is emptied (set to NaN) for the prediction range, preventing data leakage, and the input data preparation logic has been improved for better compatibility with versioned datasets. [[1]](diffhunk://#diff-538a0b7604196ca0aecc36889d00cb76cfb96910fa4593cc35c8e5979d4251f9R99-R113) [[2]](diffhunk://#diff-538a0b7604196ca0aecc36889d00cb76cfb96910fa4593cc35c8e5979d4251f9R179-R184)
* A new `median_series` method has been added to `ForecastDataset` for convenient extraction of the median forecast series.

**Model Storage and Serialization:**

* The `LocalModelStorage` implementation now saves and loads only the model state (not the whole model object), improving compatibility and robustness. [[1]](diffhunk://#diff-d566cd0d28f8066be9497bc5a7a3bcda84bcdcda80627dbdf2475b24fdf69cd8L17-R18) [[2]](diffhunk://#diff-d566cd0d28f8066be9497bc5a7a3bcda84bcdcda80627dbdf2475b24fdf69cd8L74-R83)

**Model and Hyperparameter Configuration:**

* Default hyperparameters for `GBLinearHyperParams` have been updated to more regularized and conservative values, and the `verbosity` field now accepts `True` as a valid value. [[1]](diffhunk://#diff-1725b6c0037fdfb4a0cffae2fcdfa37f98516045a969b09bf8f29b909f2fd9a5L52-R62) [[2]](diffhunk://#diff-1725b6c0037fdfb4a0cffae2fcdfa37f98516045a969b09bf8f29b909f2fd9a5L94-R94)
* The fitting logic for the GBLinear forecaster now properly includes both training and validation datasets in the evaluation set.

**Bug Fixes:**

* Loss functions now reshape sample weights correctly when there are multiple quantiles, fixing a broadcasting bug. [[1]](diffhunk://#diff-4b74bf8fb91af0e8d92f246a3a6ba11fab16e2e84a0753c65780109288efa5b6L80-R80) [[2]](diffhunk://#diff-4b74bf8fb91af0e8d92f246a3a6ba11fab16e2e84a0753c65780109288efa5b6L178-R178) [[3]](diffhunk://#diff-4b74bf8fb91af0e8d92f246a3a6ba11fab16e2e84a0753c65780109288efa5b6L260-R260)

These changes together enhance the usability, correctness, and flexibility of the forecasting pipeline, making it easier to build, train, and evaluate advanced forecasting models.

### CODE REVIEW

File: examples/examples/configuring_model_pipeline_example.py [hunk 1]
```diff
@@ -51,28 +53,55 @@
...
-        config=ConstantMedianForecasterConfig(
+    forecaster=GBLinearForecaster(
+        config=GBLinearForecasterConfig(
             horizons=[LeadTime.from_string("PT36H")],
             quantiles=[Q(0.5), Q(0.1), Q(0.9)],
+            hyperparams=GBLinearHyperParams(
+                n_estimators=1000,
+                learning_rate=0.3,
+                # reg_alpha=0.0,  # noqa: ERA001
+                # reg_lambda=1.0,  # noqa: ERA001
```
Comment (lschilders) [inline examples/examples/configuring_model_pipeline_example.py:102]: Do we want to keep these? Then I'd uncomment

**Review by lschilders — APPROVED** [review]
LGTM!

File: examples/examples/configuring_model_pipeline_example.py (hunk 1 above)
Comment (egordm) [reply examples/examples/configuring_model_pipeline_example.py:102]: Nope these are bad params just for the demo. The defaults are configured to good ones. 

I will keep them for now, just to indicate that these are not correct. Later we will restructure the test.
